### PR TITLE
Let outfit datums handle disabilities

### PR DESCRIPTION
**Pull request by ShiftyRail**
Now all I need to do is to make them handle the priority reward thing, make a unit test, and I'll hopefully done with this thing

### CODE REVIEW

File: code/datums/outfit/outfit.dm [hunk 1]
```diff
@@ -77,19 +77,33 @@
...
+/datum/outfit/proc/pre_equip_disabilities(var/mob/living/carbon/human/H, var/list/items_to_equip)
+	if (H.client.IsByondMember())
+		to_chat(H, "Thank you for supporting BYOND!")
+		items_to_collect[/obj/item/weapon/storage/box/byond] = GRASP_LEFT_HAND
+
+	if (!give_disabilities_equipment)
+		return
+	if (H.disabilities & ASTHMA)
+		items_to_collect[/obj/item/device/inhaler] = "Survival Box"
+	if (!items_to_equip[slot_glasses_str])
```
Comment (DamianX) [inline code/datums/outfit/outfit.dm:93]: `items_to_equip` seems to be a list of type paths -> location identifiers, but you're accessing it like the key is a location identifier

**Comment by DamianX** [comment]
Also it looks like you still have a few "Surival Box"es in there.

File: code/datums/outfit/outfit.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/datums/outfit/outfit.dm:93]: I realise I made a huge design mistake years ago when making the prototype, but this code is correct.
`items_to_collect` are path => location.
`items_to_equip` are a subset of `items_to_spawn` which are locations => list of paths.

I need to change that.

**Comment by ShiftyRail** [comment]
In `items_to_spawn`, the main information is the slot, with the "conditional" info being what is spawned depending on circumstances (the alt-title, etc...).

This is why it's a list of the form :
```
items_to_spawn = list(
    "Default" = list(
        slot_x_str = list(
              "Mechanic" = /path/...
        )
    )
)
```
However, in `items_to_collect`, the primary information is the path, with the associative key being the "fallback" slot. That's why I coded it this way. I think it would be more confusing to have `items_to_collect` be a list of `fallback_slot = path`, because it would make it seem the fallback slot is the important thing, not the path.

**Pull request merged**